File: Single Number II.c
class Main {
public:
    int singleNumber(vector<int>& nums) {
        int N = nums.size();
        
        int ans = 0;
        
        for(int i=0; i<=31; i++){
            int ctr = 0;
            for(int j : nums){
                if(j&(1<<i)){
                    ctr++;
                }
            }
            if(ctr%3) ans |= 1<<i;
        }
        return ans;
    }
};

File: Bit_Manipulation/Single_Number_II.java
package Bit_Manipulation;

public class Single_Number_II {
    public int singleNumber(int[] nums) {
        int result = 0, ctr = 0;
        int size = nums.length;
        for (int i = 0; i < size; i++) {
            result = (result ^ nums[i]) & (~ctr);
            ctr = (ctr ^ nums[i]) & (~result);
        }
        return result;
    }
}
